File: Geeker-Admin-QianTai/src/api/modules/menu.ts
import { PORT1 } from "@/api/config/servicePort";
import http from "@/api";

/** 与后端 build_menu_tree_all 节点一致（含 children） */
export interface MenuTreeNode {
  id: number;
  label: string;
  menuType: string;
  parentId: number | null;
  path: string;
  name: string;
  component: string;
  permission?: string;
  meta: { icon?: string; title?: string; [k: string]: unknown };
  sort?: number;
  remark?: string;
  /** 后端接口路径前缀，逗号分隔，仅 MENU 使用 */
  apiPathPrefix?: string;
  children?: MenuTreeNode[];
}

/** 菜单管理：全量树（含停用） */
export const getMenuManageTree = () => {
  return http.get<MenuTreeNode[]>(PORT1 + `/menu/manage_tree`);
};

export const addMenu = (params: {
  parentId?: number | null;
  menuType: string;
  name: string;
  title: string;
  path?: string;
  component?: string;
  icon?: string;
  permission?: string;
  sort?: number;
  remark?: string;
  apiPathPrefix?: string;
}) => {
  return http.post(PORT1 + `/menu/add`, params);
};

export const editMenu = (params: {
  id: string | number;
  parentId?: number | null;
  menuType?: string;
  name?: string;
  title?: string;
  path?: string;
  component?: string;
  icon?: string;
  permission?: string;
  sort?: number;
  remark?: string;
  status?: boolean;
  apiPathPrefix?: string;
}) => {
  return http.post(PORT1 + `/menu/edit`, params);
};

export const deleteMenu = (params: { id: string | number }) => {
  return http.post(PORT1 + `/menu/delete`, params);
};

File: backend/schemas/menu.py
from typing import Optional, Union

from pydantic import BaseModel, Field

MENU_TYPES = frozenset({"CATALOG", "MENU", "BUTTON"})


class MenuAddBody(BaseModel):
    parentId: Optional[int] = Field(None, description="父级菜单 ID，根节点不传")
    menuType: str = Field("MENU", description="CATALOG / MENU / BUTTON")
    name: str = Field(..., min_length=1, description="路由 name")
    title: str = Field(..., min_length=1, description="显示标题")
    path: Optional[str] = None
    component: Optional[str] = None
    icon: Optional[str] = None
    permission: Optional[str] = None
    sort: int = 0
    remark: Optional[str] = None
    apiPathPrefix: Optional[str] = Field(
        None,
        description="MENU 类型可选：后端接口路径前缀，多条逗号分隔，如 /api/user,/api/auth",
    )


class MenuEditBody(BaseModel):
    id: Union[str, int]
    parentId: Optional[int] = None
    menuType: Optional[str] = None
    name: Optional[str] = None
    title: Optional[str] = None
    path: Optional[str] = None
    component: Optional[str] = None
    icon: Optional[str] = None
    permission: Optional[str] = None
    sort: Optional[int] = None
    remark: Optional[str] = None
    status: Optional[bool] = None
    apiPathPrefix: Optional[str] = None


class MenuDeleteBody(BaseModel):
    id: Union[str, int] = Field(..., description="菜单 ID")
